src/backend/tty.rs

```diff
@@ -1236,6 +1236,67 @@ impl Tty {
         }
     }
 
+    pub fn get_gamma_length(&self, output: &Output) -> Option<u32> {
+        let node = output.user_data().get::<TtyOutputState>().unwrap().node;
+        let devices = self.devices.get(&node);
+        if let Some(device) = devices {
+            for (_connector, crtc) in device.drm_scanner.crtcs() {
+                let crt_info = device.drm.get_crtc(crtc).unwrap();
+                return Some(crt_info.gamma_length());
+            }
+        }
+        None
+    }
+
+    pub fn get_gamma(&self, output: &Output) -> Option<Vec<u16>> {
+        let node = output.user_data().get::<TtyOutputState>().unwrap().node;
+        let devices = self.devices.get(&node);
+        if let Some(device) = devices {
+            for (_connector, crtc) in device.drm_scanner.crtcs() {
+                let crt_info = device.drm.get_crtc(crtc).unwrap();
+                let gamma_length = crt_info.gamma_length() as usize;
+                let mut red = vec![0; gamma_length];
+                let mut green = vec![0; gamma_length];
+                let mut blue = vec![0; gamma_length];
+                device
+                    .drm
+                    .get_gamma(crtc, &mut red, &mut green, &mut blue)
+                    .unwrap_or_else(|err| warn!("error getting gamma for: {err:?}"));
```

Comment:
This `unwrap_or_else()` will not return `None` as intended. What you want is this:

```suggestion
                if let Err(err) = device.drm.get_gamma(crtc, &mut red, &mut green, &mut blue) {
                    warn!("error getting gamma for crtc {crtc:?}: {err:?}");
                    return None;
                }
```